bento/crates/api/src/helpers.rs

```diff
@@ -10,18 +10,33 @@ use workflow_common::{
     AUX_WORK_TYPE, COPROC_WORK_TYPE, EXEC_WORK_TYPE, ExecutorResp, JOIN_WORK_TYPE, PROVE_WORK_TYPE,
 };
 
+/// Extract reserved value from api_key string
+/// Format: "reserved:N" where N is a positive number (e.g., "reserved:5" or "reserved:10")
+/// Returns 0 if no reserved value is specified
+pub fn extract_reserved(api_key: &str) -> i32 {
+    if let Some(reserved_str) = api_key.strip_prefix("reserved:") {
```

Comment:
Any reason we make it so it has to be a prefix? Just feels more brittle if we ever need to use api_key/user_id to encode additional data

Reply:
Would this not be backwards compatible? Could just parse a v2 prefix if we wanted to support more in the future? What is the suggestion to do instead?